pay-api/src/pay_api/resources/v1/eft_short_names.py

```diff
@@ -0,0 +1,112 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Resource for Payment account."""
+from http import HTTPStatus
+
+from flask import Blueprint, current_app, jsonify, request
+from flask_cors import cross_origin
+
+from pay_api.exceptions import BusinessException, error_to_response
+from pay_api.schemas import utils as schema_utils
+from pay_api.services.eft_short_names import EFTShortnames as EFTShortnameService
+from pay_api.utils.auth import jwt as _jwt
+from pay_api.utils.endpoints_enums import EndpointEnum
+from pay_api.utils.enums import Role
+from pay_api.utils.errors import Error
+from pay_api.utils.trace import tracing as _tracing
+
+bp = Blueprint('EFT_SHORT_NAMES', __name__, url_prefix=f'{EndpointEnum.API_V1.value}/eft-shortnames')
+
+
+@bp.route('', methods=['GET', 'OPTIONS'])
+@cross_origin(origins='*', methods=['GET', 'POST'])
+@_jwt.requires_auth
+@_jwt.has_one_of_roles([Role.SYSTEM.value, Role.STAFF.value])
+def get_eft_shortnames():
+    """Get all eft short name records."""
+    current_app.logger.info('<get_eft_shortnames')
+
+    include_all: bool = bool(request.args.get('includeAll', 'false').lower() == 'true')
+    page: int = int(request.args.get('page', '1'))
+    limit: int = int(request.args.get('limit', '10'))
+
+    response, status = EFTShortnameService.search(include_all, page, limit), HTTPStatus.OK
+    current_app.logger.debug('>get_eft_shortnames')
+    return jsonify(response), status
+
+
+@bp.route('', methods=['POST', 'OPTIONS'])
+@cross_origin(origins='*')
+@_jwt.requires_auth
+@_jwt.has_one_of_roles([Role.SYSTEM.value, Role.STAFF.value])
+def post_eft_shortname():
+    """Create an eft short name record."""
+    current_app.logger.info('<post_eft_shortname')
+    request_json = request.get_json()
+    current_app.logger.debug(request_json)
+
+    # Validate the input request
+    valid_format, errors = schema_utils.validate(request_json, 'eft_short_name')
+
+    if not valid_format:
+        return error_to_response(Error.INVALID_REQUEST, invalid_params=schema_utils.serialize(errors))
+    try:
+        response, status = EFTShortnameService.create(request_json), HTTPStatus.CREATED
+    except BusinessException as exception:
+        return exception.response()
+    current_app.logger.debug('>post_eft_shortname')
+    return jsonify(response.asdict()), status
+
+
+@bp.route('/<int:short_name_id>', methods=['GET', 'OPTIONS'])
+@cross_origin(origins='*', methods=['GET', 'PATCH'])
+@_tracing.trace()
+@_jwt.requires_auth
+@_jwt.has_one_of_roles([Role.SYSTEM.value, Role.STAFF.value])
+def get_eft_shortname(short_name_id: int):
+    """Get EFT short name details."""
+    current_app.logger.info('<get_eft_shortname')
+    eft_short_name = EFTShortnameService.find_by_short_name_id(short_name_id)
+
+    if eft_short_name is None:
```

Comment:
`if not (short_name := EFTShortnameService.find_by_short_name_id(short_name_id))`